components/camera/client_test.go

```diff
@@ -180,15 +178,14 @@ func TestClient(t *testing.T) {
 		test.That(t, err, test.ShouldBeNil)
 		camera1Client, err := camera.NewClientFromConn(context.Background(), conn, "", camera.Named(testCameraName), logger)
 		test.That(t, err, test.ShouldBeNil)
-		ctx := gostream.WithMIMETypeHint(context.Background(), rutils.MimeTypeRawRGBA)
-		frame, _, err := camera.ReadImage(ctx, camera1Client)
+		frame, err := camera.ImageFromVideoSource(context.Background(), rutils.MimeTypeRawRGBA, nil, camera1Client)
 		test.That(t, err, test.ShouldBeNil)
 		compVal, _, err := rimage.CompareImages(img, frame)
 		test.That(t, err, test.ShouldBeNil)
 		test.That(t, compVal, test.ShouldEqual, 0) // exact copy, no color conversion
-		imageReleasedMu.Lock()
-		test.That(t, imageReleased, test.ShouldBeTrue)
-		imageReleasedMu.Unlock()
+		_, err = camera.ImageFromVideoSource(context.Background(), rutils.MimeTypeRawRGBA, map[string]interface{}{"empty": true}, camera1Client)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "received empty bytes from Image method")
```

Comment:
```suggestion
		test.That(t, err, test.ShouldBeError, errors.New("received empty bytes from Image method"))
```

Reply:
```
    --- FAIL: TestClient/camera_client_1 (0.00s)
        /Users/seanyu/Projects/sean-rdk/components/camera/client_test.go:189: Expected: 'received empty bytes from Image method'
            Actual:   'could not get image bytes from camera: rpc error: code = Unknown desc = received empty bytes from Image method of camera1'
            (Should be equal)
```
After the patch... similar to https://github.com/viamrobotics/rdk/pull/4487#discussion_r1843887777

Open to suggestions to unwrap somehow?

Reply:
does `test.ShouldWrap` work?

Ultimately, we need to be able to assert what errors happened during tests without relying on exact substring matches.

Reply:
alright, then this can stay as is. 

Thanks for explaining